src/libs/Sound/index.ts

```diff
@@ -1,71 +1,75 @@
-import Onyx from 'react-native-onyx';
-import Sound from 'react-native-sound';
+import {Howl} from 'howler';
 import type {ValueOf} from 'type-fest';
-import ONYXKEYS from '@src/ONYXKEYS';
+import Log from '@libs/Log';
+import {getIsMuted, SOUNDS, withMinimalExecutionTime} from './BaseSound';
 import config from './config';
 
-let isMuted = false;
-
-Onyx.connect({
-    key: ONYXKEYS.USER,
-    callback: (val) => (isMuted = !!val?.isMutedAllSounds),
-});
+function cacheSoundAssets() {
+    // Exit early if the Cache API is not available in the current browser.
+    if (!('caches' in window)) {
+        return;
+    }
 
-const SOUNDS = {
-    DONE: 'done',
-    SUCCESS: 'success',
-    ATTENTION: 'attention',
-    RECEIVE: 'receive',
-} as const;
+    caches.open('sound-assets').then((cache) => {
+        const soundFiles = Object.values(SOUNDS).map((sound) => `${config.prefix}${sound}.mp3`);
 
-/**
- * Creates a version of the given function that, when called, queues the execution and ensures that
- * calls are spaced out by at least the specified `minExecutionTime`, even if called more frequently. This allows
- * for throttling frequent calls to a function, ensuring each is executed with a minimum `minExecutionTime` between calls.
- * Each call returns a promise that resolves when the function call is executed, allowing for asynchronous handling.
- */
-function withMinimalExecutionTime<F extends (...args: Parameters<F>) => ReturnType<F>>(func: F, minExecutionTime: number) {
-    const queue: Array<[() => ReturnType<F>, (value?: unknown) => void]> = [];
-    let timerId: NodeJS.Timeout | null = null;
+        // Cache each sound file if it's not already cached.
+        const cachePromises = soundFiles.map((soundFile) => {
+            return cache.match(soundFile).then((response) => {
+                if (response) {
+                    return;
+                }
+                return cache.add(soundFile);
+            });
+        });
```

Comment:
@rayane-djouah  Not sure about this suggestion, but is it better to use in-memory cache if we want to update the sound assets on each page load?

To optimize performance, consider clearing sound assets during logout and resetting the app on the troubleshooting page.

src/libs/Sound/index.ts
```tsx

function clearSoundAssetsCache() {
    // Exit early if the Cache API is not available in the current browser.
    if (!('caches' in window)) {
        return;
    }

    caches.delete('sound-assets').then((success) => {
        if (success) {
            console.log('Sound assets cache cleared.');
        } else {
            console.log('Failed to clear sound assets cache.');
        }
    }).catch((error) => {
        console.error('Error clearing sound assets cache:', error);
    });
}

```

And execute that function in:

src/libs/actions/App.ts
```tsx

function clearOnyxAndResetApp(shouldNavigateToHomepage?: boolean) {
......
    clearSoundAssetsCache();
.....

```

And for clear action in Troubleshoot page:

src/libs/actions/App.ts
```tsx

function clearOnyxAndResetApp(shouldNavigateToHomepage?: boolean) {
.........
    clearSoundAssetsCache();
}

```


https://github.com/user-attachments/assets/11414c63-0d61-435d-b09d-7f4675c4495c

Reply:
@rayane-djouah how would we determine version? usually server will send the version to invalidate the cache. Is there any API for that purpose?

Reply:
Another common approach is version the file name sound-v1.mp3, since the name already sotred as key to put the cache I think we don't need any other change to version the cache in FE

Reply:
The updated main has been merged and clearSoundAssetsCache has been implemented